os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -99,6 +105,31 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 	abort_mode = true;
 #endif
 
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+	struct watchdog_status_s wd_status;
+	bool is_wd_paused = false;
+	mfdbg("Temporarily pause active watchdog\n");
+	int fd = open(CONFIG_WATCHDOG_DEVPATH, O_RDWR);
+	if (fd < 0) {
+		mfdbg("Fail to open %s, errno %d\n", CONFIG_WATCHDOG_DEVPATH, get_errno());
+	} else {
+		/* if watchdoc driver is successfully opened, try to get watchdog state */
+		if (ioctl(fd, WDIOC_GETSTATUS, (unsigned long)&wd_status) != OK) {
+			mfdbg("Fail to get watchdog state, errno %d\n", get_errno());
+
+		/* if watchdog is running, pause the watchdog */
+		} else if (wd_status.flags & WDFLAGS_ACTIVE && !(wd_status.flags & WDFLAGS_WAIT)) {
+			if (ioctl(fd, WDIOC_PAUSE, 0) != OK) {
+				mfdbg("Fail to pause watchdog, errno %d\n", get_errno());
```

Comment:
Is just printing an error message sufficient here? We might run into watchdog reset again in this case, right?

Reply:
yes, this case can watchdog reset.
but I think, in fail case is not a normal operation.

even if pause fail, we need to print heapinfo log as many as possible that is next code.
so, I didn't add any action

if must be paused watchdog in error operation, I can find way without watchdog driver.